File: competitive-programming/1.Math/1.Number Theory/2.Sieve of Eretosthenes/1.sol.py

def sieve(num):
    is_prime = [1 for i in range(num+1)]
    is_prime[0] = 0
    is_prime[1] = 0

    for i in range(int(num**0.5)+1):
        if is_prime[i]:
            for j in range(i*i, num+1, i):
                is_prime[j] = 0
    return is_prime

num = 100
result = [idx for idx, i in enumerate(sieve(num)) if i==1]
print(result)

File: competitive-programming/1.Math/1.Number Theory/2.Sieve of Eretosthenes/1.sol.go
package main

import "fmt"

func sieve(num int) []bool {
	primes := make([]bool, num+1)

	for i := 0; i <= num; i++ {
		if i < 2 {
			primes[i] = false
			continue
		}
		primes[i] = true
	}
	p := 2
	for p*p <= num {
		if primes[p] == true {
			for i := p * p; i <= num; i += p {
				primes[i] = false
			}
		}
		p++
	}
	return primes
}

func main() {
	num := 100
	primes := sieve(num)
	fmt.Println("Primes up to", num, ":")
	for idx, prime := range primes {
		if prime == true {
			fmt.Print(idx, " ")
		}
	}
}
